aten/src/ATen/native/cpu/FlashAttentionKernel.cpp

```diff
@@ -21,6 +21,53 @@ namespace at::native {
 
 namespace {
 
+// out = val * a + b
+template <typename T1, typename T2>
+inline void _scale_attn_mask_fusion_kernel(
+    T1* a,
+    T2* b,
+    const int& size,
+    T1* out,
+    T1& val) {
+  auto vec_size = at::vec::Vectorized<T1>::size();
+  auto vec_scale = at::vec::Vectorized<T1>(val);
+  for (long i = 0; i < vec_size * (size / vec_size); i += vec_size) {
+    auto tmp0 = at::vec::Vectorized<T1>::loadu(a + i);
+    auto tmp1 = at::vec::Vectorized<T2>::loadu(b + i);
+    auto tmp2 = at::vec::convert<T1>(tmp1);
+    auto tmp3 = tmp0 * vec_scale + tmp2;
+    _store(out + i, tmp3);
+  }
+  for (long i = vec_size * (size / vec_size); i < size; i++) {
+    auto tmp0 = a[i];
+    auto tmp1 = (T1) b[i];
+    out[i] = tmp0 * val + tmp1;
+  }
+}
+
+// out = val * a + b
+template <typename T1>
+inline void _scale_attn_mask_fusion_kernel(
```

Comment:
NIT: Maybe you don't have to add this one and can reuse the function above. You can make `vec::convert` on the same data type as a nop (by adding a specialization to `vec_convert.h`).

Reply:
Now, vec::convert can support with the same data type as a noop.